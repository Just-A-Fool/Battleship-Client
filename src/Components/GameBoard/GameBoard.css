.message {
  color: red;
}

.errorMessage {
  color: red;
}

p {
  font-size: 1.2rem;
}

.grid {
  width: 90%;
  margin: 0 auto;
  margin-bottom: 2rem;
}

.gameroom {
  margin-top: 8rem;
}

@media screen and (min-width: 1025px) {
  .grid-box {
    display: flex;
    justify-content: center;
  }

  .grid {
    width: 45%;
    margin: 0 1rem;
    margin-bottom: 2rem;
  }

  p {
    font-size: 1.5rem;
  }
  
}
