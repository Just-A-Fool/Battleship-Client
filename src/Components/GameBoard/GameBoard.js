import React from 'react';
import UserGrid from '../UserGrid/UserGrid';
import io from 'socket.io-client';
import config from '../../config';
import OpponentGrid from '../OpponentGrid/OpponentGrid';
import BattleShipContext from '../../Contexts/battleship-context';
import './GameBoard.css';
import TokenService from '../../Services/token-service';




class GameBoard extends React.Component {

  /*
  
    export object = {
        hits
        misses
        our Ships
    };

    import object = {
        our hits 
        our misses
        opponents hits/misses
        our ships
        whose turn
    };
  */ 


  /*
  Add opponents hits/misses to state
  */
  state = {
    userShips: [],
    opponentShips: [],
    userHits: [],
    userMisses: [],
    opponentHits: [],
    opponentMisses: [],
    userTurn: false,
    //hard coding gameId and playerNum for testing purposes temporarily
    gameId: 1,
    playerNum: 'player1',
    error: null,
    room: this.props.room,
    socket: null
  }

  //can we move this to a separate context provider file?
  setError = (err) => {
    this.setState({
      error: err.error
    })
  }

  clearError = () => { 
    this.setState({ error: null, }) 
  }
  
  componentDidMount = () =>{
    //fetch game data based on game id. set the state with the game data and pass
    //down as props to userGrid (needs ships for the user and opponent hits) and opponentGrid
    //(needs user's hits and misses to re-mark the board)
    const socket = io(config.API_ENDPOINT, {
      transportOptions: {
        polling: {
          extraHeaders: {
            'Authorization' : `Bearer ${TokenService.getAuthToken()}`
          }
        }
      }
    });
    let roomName = this.state.room ? this.state.room: 'random';
    socket.emit('join_room', roomName);

    
    socket.on('joined', data =>  {
      console.log(data);
      this.setState({
        playerNum: data.player,
        room: data.room,
        userTurn: data.player === 'player1' ? true: false,
        gameId: data.gameId,
        socket: socket
      })
    });

    socket.on('reconnected', () => {
      this.setState({
        socket:socket
      })
    });
  }

  render () {
    return (
      <BattleShipContext.Provider value={{
        gameId: this.state.gameId,
        playerNum: this.state.playerNum,
        error: this.state.error,
        setError: this.setError,
        clearError: this.clearError
      }}>
      <>
      {this.state.error && <p className='errorMessage'>{this.state.error}</p>}
      <h2>Your Ships</h2>
      {this.state.socket && <UserGrid socket={this.state.socket} /> }

      <h2>Opponent Ships</h2>
      <OpponentGrid socket={this.state.socket} room={this.state.room}/>
      </>
      </BattleShipContext.Provider>
    )
  };
};

export default GameBoard;


