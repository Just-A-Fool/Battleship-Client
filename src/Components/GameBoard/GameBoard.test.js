import React from 'react';
import ReactDOM from 'react-dom';
import GameBoard from './GameBoard';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme'


describe('GameBoard component', () => {
  let gameData = {
      currentUser: null,
      gameId: null,
      opponentHits: [],
      opponentMisses: [],
      opponentShips: null,
      room_id: null,
      turn: null,
      userHits: [],
      userMisses: [],
      userShips: [],
      shipTileValues: [],
      resumedGame: null,
      shipsReady: null,
    }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><GameBoard gameData={gameData}/></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})