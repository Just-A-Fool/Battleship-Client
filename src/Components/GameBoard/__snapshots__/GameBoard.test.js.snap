// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard component Displays the Gameboard component (Container for UserGrid, OpponentGrid, Chat) 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(GameBoard)
      gameData={
        Object {
          "currentUser": null,
          "gameId": null,
          "opponentHits": Array [],
          "opponentMisses": Array [],
          "opponentShips": null,
          "resumedGame": null,
          "room_id": null,
          "shipTileValues": Array [],
          "shipsReady": null,
          "turn": null,
          "userHits": Array [],
          "userMisses": Array [],
          "userShips": Array [],
        }
      }
    >
      <Route>
        <GameBoard
          gameData={
            Object {
              "currentUser": null,
              "gameId": null,
              "opponentHits": Array [],
              "opponentMisses": Array [],
              "opponentShips": null,
              "resumedGame": null,
              "room_id": null,
              "shipTileValues": Array [],
              "shipsReady": null,
              "turn": null,
              "userHits": Array [],
              "userMisses": Array [],
              "userShips": Array [],
            }
          }
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div
            className="gameroom"
          >
            <withRouter(Header)>
              <Route>
                <Header
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    <Banner>
                      <div
                        className="banner"
                      >
                        <h1>
                          BATTLESHIP
                        </h1>
                      </div>
                    </Banner>
                    <header
                      className="main-header"
                    >
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Dashboard
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Help
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                          Logout
                        </button>
                      </ForwardRef>
                    </header>
                  </div>
                </Header>
              </Route>
            </withRouter(Header)>
            <h2
              className="versusHeader"
            >
              Waiting for Opponent...
            </h2>
            <div
              className="progress"
            >
              <h4>
                Progress &gt;&gt;
              </h4>
              <ul
                className="shipCounter"
              />
            </div>
            <div
              className="grid-box"
            />
            <ForwardRef
              onClick={[Function]}
            >
              <button
                className="Button"
                onClick={[Function]}
              >
                Back
              </button>
            </ForwardRef>
            <Footer>
              <footer>
                Copyright Â© since 2020
              </footer>
            </Footer>
          </div>
        </GameBoard>
      </Route>
    </withRouter(GameBoard)>
  </Router>
</BrowserRouter>
`;
